src/coreclr/runtime/amd64/AllocFast.S

```diff
@@ -233,11 +233,11 @@ LEAF_END RhpNewArrayFast, _TEXT
 LEAF_ENTRY RhpNewPtrArrayFast, _TEXT
```

Comment:
The comment on this method is incorrect. The element count is in 64-bit RSI register, not in the 32-bit ESI register (this bug is in multiple places - not commenting on all of them).